callout/src/main/java/com/google/apigee/ProtoMessageBuilders/ExecutionContextProtoMessageBuilder.java

```diff
@@ -0,0 +1,142 @@
+                      .map(this::buildFault)
+                      .collect(Collectors.toList()));
+    }
+    return executionContextBuilder.build();
+  }
+
+  /**
+   * Gets the appropriate FlowType enumerator from the {@link ExecutionContext}. Used to indicate an
+   * error or request flow.
+   *
+   * @param executionContext {@link ExecutionContext} from which to extract the flow type.
+   * @return ExecutionContext.FlowType enumerator value
+   */
+  private ExecutionOuterClass.ExecutionContext.FlowType getFlowType(
+      ExecutionContext executionContext) {
+    if (executionContext.isErrorFlow()) {
+      return ExecutionOuterClass.ExecutionContext.FlowType.ERROR;
+    }
+    if (executionContext.isRequestFlow()) {
+      return ExecutionOuterClass.ExecutionContext.FlowType.REQUEST;
+    }
+    return ExecutionOuterClass.ExecutionContext.FlowType.UNKNOWN_FLOW_TYPE;
+  }
+
+  /**
+   * Builds a Fault Protocol Buffer Message using a {@link Fault} object.
+   *
+   * @param fault {@link Fault} used to construct the Protocol Buffer Message.
+   * @return Fault Protocol Buffer Message
+   */
+  private ExecutionOuterClass.ExecutionContext.Fault buildFault(Fault fault) {
+    return ExecutionOuterClass.ExecutionContext.Fault.newBuilder()
+        .setCategory(getFaultCategory(fault))
+        .setSubCategory(fault.getSubCategory())
+        .setName(fault.getName())
+        .setReason(fault.getReason())
+        .putAllAttributes(buildAttributesMap(fault))
+        .build();
+  }
+
+  /**
+   * Determines the appropriate Category for the Fault Protocol Buffer Message
+   *
+   * @param fault Fault from which to extract the Category
+   * @return Fault.Category Protocol Buffer enumerator value
+   */
+  private ExecutionOuterClass.ExecutionContext.Fault.Category getFaultCategory(Fault fault) {
+    switch (fault.getCategory()) {
+      case Messaging:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.MESSAGING;
+      case Step:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.STEP;
+      case Transport:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.TRANSPORT;
+      case System:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.SYSTEM;
+      default:
+        return ExecutionOuterClass.ExecutionContext.Fault.Category.UNKNOWN_CATEGORY;
+    }
+  }
+
+  /**
+   * Constructs the String to Any attributes map for a Fault protocol buffer message
+   *
+   * @param fault Fault from which attributes map is extracted from
+   * @return Map of String to Any for protocol buffer message
+   */
+  private Map<String, Any> buildAttributesMap(Fault fault) {
+    Map<String, Any> attributesMap = new HashMap<>();
+    fault
+        .getAttributes()
+        .forEach(
+            (key, value) -> {
+              try {
+                attributesMap.put(key, Any.parseFrom(objectToByteArray(value)));
+              } catch (Exception exception) {
```

Comment:
This catch-block effectively silences the exception.  Are there any downsides to that?

Reply:
changed this to throw an uncheckedioexception